### ISSUE TITLE
ppocr/modeling/det_resnet.py从paddlepaddle转pytorch的代码是否方便提供

### ISSUE THREAD

**Issue by bad-meets-joke** [WenmuZhou/PytorchOCR#303]
背景：最近想复现DB++在ICDAR2015上的结果，然后在自己数据集上训练。

问题：训练报错，缺少resnet到pytorch框架的适配，目前只有resnet_vd的适配，请问何时提供

额外问题：DB++与PP-OCR v4相比，能力谁更强

**Comment by WenmuZhou** [comment]
自己每个算子转的，没用代码

**Comment by bad-meets-joke** [comment]
嗯嗯，这份代码目前是不是还不支持DB++在ICDAR2015上的结果复现，主要因为
（1）torchocr/modeling/缺少了文件det_resnet.py来定义backbone ResNet
（2）configs对应的配置文也需要做很多修改

关于（1），我从 https://github.com/frotms/PaddleOCR2Pytorch 拿来了det_resnet.py。
关于（2）我也完成了一些名称修改，从而能正常读取配置文件。
此外，我也把PaddleOCR提供的预训练权重通过你的权重转换代码成功完成转换，输出误差为1.7505917e-08

现在我能正常开启训练了，但**不能复现结果，存在较大差距**
目前结果为Precision=88.83 ｜ Recall=78.86 ｜  Hmean=83.55
原文结果为Precision=90.89 ｜ Recall=82.66 ｜  Hmean=86.58

不知道是哪里导致这样的差异？

**Comment by WenmuZhou** [comment]
看看配置对不对

**Comment by bad-meets-joke** [comment]
为了正常运行，我修改了配置文件configs/det/det_r50_db++_icdar15.yml的格式，但在内容上与你的保持一致，具体如下：

Global:
  debug: false
  use_gpu: true
  device: gpu  # 新增
  epoch_num: 1000
  log_smooth_window: 20
  print_batch_step: 10
  output_dir: ./output/debug/  # ./output/det_r50_icdar15/
  save_epoch_step: 100
  eval_epoch_step: [ 0, 20 ]
  cal_metric_during_train: false
  pretrained_model: ./pretrain_models/ResNet50_dcn_asf_synthtext_pretrained.pth
  checkpoints: null
  save_inference_dir: null
  use_visualdl: false
  use_tensorboard: false  # 新增！
  infer_img: doc/imgs_en/img_10.jpg
  save_res_path: ./checkpoints/det_db/predicts_db.txt

Architecture:
  model_type: det
  algorithm: DB++
  Transform: null
  Backbone:
    name: ResNet
    layers: 50
    dcn_stage: [False, True, True, True]
  Neck:
    name: DBFPN
    out_channels: 256
    use_asf: True
  Head:
    name: DBHead
    k: 50

Loss:
  name: DBLoss
  balance_loss: true
  main_loss_type: BCELoss
  alpha: 5
  beta: 10
  ohem_ratio: 3

Optimizer:
  name: SGD  # Momentum
  momentum: 0.9
  lr: 0.007
  weight_decay: 0.0001

LRScheduler:
  name: PolynomialLR  #DecayLearningRate
  power: 0.9
  lr_end: 0

PostProcess:
  name: DBPostProcess
  thresh: 0.3
  box_thresh: 0.6
  max_candidates: 1000
  unclip_ratio: 1.5
  box_type: 'quad'

Metric:
  name: DetMetric
  main_indicator: hmean

Train:
  dataset:
    name: SimpleDataSet
    data_dir: xxxx
    label_file_list:                                                                                            
    - xxx
    ratio_list:                                                                                                 
    - 1.0 
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - DetLabelEncode: null
    - IaaAugment:
        augmenter_args:
        - type: Fliplr
          args:
            p: 0.5
        - type: Affine
          args:
            rotate:
            - -10
            - 10
        - type: Resize
          args:
            size:
            - 0.5
            - 3
    - EastRandomCropData:
        size:
        - 640
        - 640
        max_tries: 10
        keep_ratio: true
    - MakeShrinkMap:
        shrink_ratio: 0.4
        min_text_size: 8
    - MakeBorderMap:
        shrink_ratio: 0.4
        thresh_min: 0.3
        thresh_max: 0.7
    - NormalizeImage:
        scale: 1./255.
        mean:
        - 0.48109378172549
        - 0.45752457890196
        - 0.40787054090196
        std:
        - 1.0
        - 1.0
        - 1.0
        order: hwc
    - ToCHWImage: null
    - KeepKeys:
        keep_keys:
        - image
        - threshold_map
        - threshold_mask
        - shrink_map
        - shrink_mask
  loader:
    shuffle: true
    drop_last: false
    batch_size_per_card: 4
    num_workers: 8

Eval:
  dataset:
    name: SimpleDataSet
    data_dir: xxx/icdar2015/text_localization
    label_file_list:                                                                                            
    - xxx/icdar2015/text_localization/test_icdar2015_label.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - DetLabelEncode: null
    - DetResizeForTest:
        image_shape:
        - 1152
        - 2048
    - NormalizeImage:
        scale: 1./255.
        mean:
        - 0.48109378172549
        - 0.45752457890196
        - 0.40787054090196
        std:
        - 1.0
        - 1.0
        - 1.0
        order: hwc
    - ToCHWImage: null
    - KeepKeys:
        keep_keys:
        - image
        - shape
        - polys
        - ignore_tags
  loader:
    shuffle: false
    drop_last: false
    batch_size_per_card: 1
    num_workers: 2

**Comment by WenmuZhou** [comment]
里面的配置和paddle那边一样吗
